### DISCUSSION TITLE
Link to a website with responsive voice

### DISCUSSION THREAD

**Discussion by alecostella** [mozilla/hubs#4074] [Spoke Questions] [answered]
Hi, I've started to work with Spoke (very) recently, and I'm having very specific troubles with a link.

The context: I'm trying to insert a "full preview" of a web page inside a scene, but some elements won't appear in the page's preview in Hubs. Specifically, the website is this one: https://www.arkanianshenron.com/

The page consists of:
1) A widget that captures the latest tweet from a specific account (this part shows in the preview in Hubs)
2) A short looping video as a background
3) An automatic voice reading the twitter, implemented through Responsive Voice (https://responsivevoice.org/). Responsive voice asks for permission to play through a pop-up, and of course this can't happen from a Hubs previews.

The state of the problem:
1) The looping video is not showing, hence the background is a blank page
2) The voice isn't playing at all from the preview (unsurprisingly)

The desired results:
1) The looping video showing correctly in the preview
2) The auto-reading voice playing form the preview

The html the developer used is very short and easily readable. I am also quite new to html (html5 in this case), but I'm positive the solution is in the html more than it is in Hubs. Even tho I didn't make the website, I do have access to the html for modifications. Is it possible to make this work?

Thanks in advance!

**Comment by johnshaughnessy** [comment, marked as answer]
The way url embeds work in Hubs is that a backend service connects to the page, captures a screenshot, and then uses that image as a preview of the page. Unfortunately, there's no way to play sounds or to ensure that all visual elements (like videos) on the page will be correctly displayed. (The latter problem is because the software we use to capture the screenshot runs some version of a browser we have little insight into or control over.)

There has been some experimentation with bringing iframes into the 3D scene. You can read a bit about this in this blog post https://blog.mozvr.com/behind-the-scenes-at-hubs-hack-week/ and see a demo in this twitch recording https://www.twitch.tv/videos/934727861?filter=all&sort=time . But iframes are not yet added to Hubs and it's not clear when or if they will be.

To embed this content into hubs, you'll need to either take a screenshot or video of what you want to display and use it as a static piece of content.
